src/MainPage.jsx

```diff
@@ -0,0 +1,144 @@
+// MainPage.js
+
+import React, { useState } from 'react';
+import './MainPage.css'; // Import the CSS file for styles
+
+function MainPage() {
+    const [collapsed, setCollapsed] = useState(false);
+
+    const toggleSidebar = () => {
+        setCollapsed(!collapsed);
+    };
+
+    const sidebarStyle = {
```

Comment:
Please separate out your styles into a separate file and use styled-components syntax. This is close, but not quite right yet.